### DISCUSSION TITLE
Does set silently swallow exceptions?

### DISCUSSION THREAD

**Discussion by vnugent** [pmndrs/zustand#1929] [Q&A] [answered]
Took me several hours to track this down.  I made a mistake mutating an array in set() but never realized it.  The action just completes without changing the state.

Does zustand silently swallow exceptions in `set()`?  Is there a way to tell it not to, at least during development.

```javascript
const useStore = create<State & Action>((set) => ({
  firstName: '',
  lastName: '',
  updateFirstName: (firstName) => set(() => {
     // accidentally cause an exception 
     throw new Error("my mistake")
     return ({ firstName: firstName }))
   }
}))

```

**Comment by dai-shi** [comment, marked as answer]
It seems to work as expected: https://codesandbox.io/s/jolly-jang-hny4qp?file=/src/App.js

<img width="351" alt="image" src="https://github.com/pmndrs/zustand/assets/490574/7265e19d-ac34-41fb-aca1-51893817e744">

**Reply by vnugent** [reply]
My bad.  I catch the exception higher up in the stack.
